### ISSUE TITLE
java.util.ServiceConfigurationError: io.opentelemetry.javaagent.shaded.io.opentelemetry.metrics.spi.MeterProviderFactory: io.opentelemetry.sdk.metrics.spi.MeterProviderFactorySdk not a subtype

### ISSUE THREAD

**Issue by anuraaga** [open-telemetry/opentelemetry-java-instrumentation#1114]
Updating my agent distribution and I see this weird configuration error. There's a good chance it's something wrong with my build scripts but copying here for now just in case.

https://github.com/anuraaga/aws-opentelemetry-java-instrumentation/pull/5/files

```
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: [opentelemetry.auto.trace 2020-08-27 11:27:55:851 +0000] [main] ERROR io.opentelemetry.javaagent.bootstrap.AgentInitializer - Throwable thrown while installing the agent
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: java.lang.reflect.InvocationTargetException
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.startAgent(AgentInitializer.java:154)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.initialize(AgentInitializer.java:59)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.OpenTelemetryAgent.agentmain(OpenTelemetryAgent.java:75)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at com.softwareaws.xray.opentelemetry.agentbootstrap.AwsAgentBootstrap.agentmain(AwsAgentBootstrap.java:32)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at com.softwareaws.xray.opentelemetry.agentbootstrap.AwsAgentBootstrap.premain(AwsAgentBootstrap.java:24)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: Caused by: java.util.ServiceConfigurationError: io.opentelemetry.javaagent.shaded.io.opentelemetry.metrics.spi.MeterProviderFactory: io.opentelemetry.sdk.metrics.spi.MeterProviderFactorySdk not a subtype
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/java.util.ServiceLoader.fail(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/java.util.ServiceLoader$2.hasNext(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/java.util.ServiceLoader$3.hasNext(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.shaded.io.opentelemetry.OpenTelemetry.loadSpi(OpenTelemetry.java:234)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.shaded.io.opentelemetry.OpenTelemetry.<init>(OpenTelemetry.java:208)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.shaded.io.opentelemetry.OpenTelemetry.getInstance(OpenTelemetry.java:194)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.shaded.io.opentelemetry.OpenTelemetry.getTracerProvider(OpenTelemetry.java:71)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:93)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:64)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	... 19 more
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: ERROR io.opentelemetry.javaagent.OpenTelemetryAgent
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: java.lang.reflect.InvocationTargetException
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.OpenTelemetryAgent.agentmain(OpenTelemetryAgent.java:75)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at com.softwareaws.xray.opentelemetry.agentbootstrap.AwsAgentBootstrap.agentmain(AwsAgentBootstrap.java:32)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at com.softwareaws.xray.opentelemetry.agentbootstrap.AwsAgentBootstrap.premain(AwsAgentBootstrap.java:24)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(Unknown Source)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: Caused by: java.lang.IllegalStateException: Agent should have been started already
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.installAgentTracer(AgentInitializer.java:164)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.initialize(AgentInitializer.java:86)
[tc-okhttp-stream-587665229] INFO io.awsobservability.instrumentation.smoketests.runner.SpringBootSmokeTest - STDERR: 	... 13 more
```

**Comment by trask** [comment]
My first guess would be your shading configuration needs to be updated because we changed shading in this repo, e.g.

```
io.opentelemetry.javaagent.shaded.io.opentelemetry.metrics.spi.MeterProviderFactory
```

i think used to be 

```
io.opentelemetry.auto.shaded.io.opentelemetry.metrics.spi.MeterProviderFactory
```

**Comment by anuraaga** [comment]
This was something that showed up during random attempts to fix what I was seeing in #1119 but is itself not a real issue (was incorrect shading config indeed). Closing
